Sources/DatadogSDKTesting/DDTestObserver.swift

```diff
@@ -126,9 +126,19 @@ class DDTestObserver: NSObject, XCTestObservation {
         state = .suite(suite: module.suiteStart(name: testSuite.name), inside: parent)
         
         if let itr = DDTestMonitor.instance?.itr {
-            let skippableTests = itr.skippableTests.filter { $0.suite == testSuite.name }.map { "-[\(testSuite.name) \($0.name)]" }
-            tests = tests.map {
-                skippableTests.contains($0.name) ? SkippedTest(for: $0) : $0
+            let skippableTests = itr.skippableTests
+            var unskippableCache: [ObjectIdentifier: UnskippableMethodChecker] = [:]
+            tests = tests.map { test in
+                guard let match = Self.testNameRegex.firstMatch(in: test.name, range: NSRange(0..<test.name.count)),
+                      let suiteRange = Range(match.range(at: 1), in: test.name),
+                      let funcRange = Range(match.range(at: 2), in: test.name) else { return test }
```

Comment:
nit: maybe worth breaking the line after {

On the first glance I got confused with formatting